src/subcommand.rs

```diff
@@ -258,7 +258,31 @@ impl Subcommand {
   fn format(config: &Config, ast: Ast, search: &Search) -> Result<(), Error<'static>> {
     config.require_unstable("The `--fmt` command is currently unstable.")?;
 
-    if let Err(io_error) =
+    if config.dry_run {
+      use std::io::Read as _;
+      let mut current_content = String::new();
+      let mut file = File::open(&search.justfile).map_err(|e| Error::Load {
+        path: search.justfile.clone(),
+        io_error: e,
+      })?;
+      file
+        .read_to_string(&mut current_content)
+        .map_err(|e| Error::Load {
+          path: search.justfile.clone(),
+          io_error: e,
+        })?;
+
+      if current_content != format!("{}", ast) {
+        Err(Error::NotWellFormatted {
```

Comment:
I think we probably shouldn't call justfiles that produce diffs "not well formatted". The just have some different formatting which is not necessarily bad, just nonstandard.

```suggestion
        Err(Error::FormatCheckFoundDiff {
```

Reply:
Good wording, applied